pkg/kubelet/kubelet.go

```diff
@@ -2212,6 +2219,16 @@ func (kl *Kubelet) fastStatusUpdateOnce() {
 	}
 }
 
+func (kl *Kubelet) deletePodSandbox(podID types.UID) {
+	if podStatus, err := kl.podCache.Get(podID); err == nil {
+		toKeep := 1
```

Comment:
do we need to do pruning here?   Or should we simplify everything so we just handle the case where we are deleting everything?

Reply:
i didn't understand the comment exactly but the logic here is from kuberuntime gc https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kuberuntime/kuberuntime_gc.go#L314

Reply:
Basically, if the pod is still running, we clean up all sandboxes except the one that is running.  If it isn't, we remove them all.